File: 프로그래머스/1/42862. 체육복/체육복.py
def solution(n, lost, reserve):
    answer = 0
    safe = n
    lost.sort()
    
    temp = []
    # 여분 있는 사람이 도난당함
    for i in range(len(lost)):
        if lost[i] in reserve:
            reserve.remove(lost[i])
            lost[i] = -1
            
            
    
    #빌리기
    for i in lost:
        if i == -1:
            continue
        # 앞뒤사람에게 빌릴수 있음
        elif i-1 in reserve:
            reserve.remove(i-1)
        elif i+1 in reserve:
            reserve.remove(i+1)
        else:
            safe -= 1    
    return safe

File: 프로그래머스/1/42862. 체육복/체육복.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n;
        List<int> losts = new List<int>(lost);
        List<int> reserves = new List<int>(reserve);
        losts.Sort();
        
        // 가지고 있는 본인이 도난당한 경우 reserves 에서 제거
        int m = losts.Count;
        for(int i = 0; i < m; i++){
            if(reserves.Contains(losts[i])){
                reserves.Remove(losts[i]);
                losts[i] = 0;
            }
        }
        
        foreach(int j in losts){
            // Console.WriteLine(j);
            if(j == 0)   continue;
            
            if(reserves.Contains(j-1)){
                reserves.Remove(j-1);
            }
            else if(reserves.Contains(j+1)){
                reserves.Remove(j+1);
            }
            else answer--;
                
        }

        return answer;
    }
}